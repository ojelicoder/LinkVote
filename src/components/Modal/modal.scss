@import '../../styles/variables';

.modal{
    &_backdrop{
        background: rgba(0, 0, 0, .65);
        bottom: 0;
        left: 0;
        overflow: auto;
        position: fixed;
        right: 0;
        top: 0;
        z-index: 1;
    }

    &_container{
        background-color: $color-mercury;
        max-width: 100%;
        margin: 150px auto;
        width: 500px;
    }
    &_header{
        margin: 0;
        display: flex;
        flex-direction: row;
        justify-content: space-between;
        background-color: $color-black;
        color: $color-white;
        padding: 10px 20px;

        &-title{
            margin: 0;
            font-weight: 500;
        }
    }
    &_close{
        background: transparent;
        color: $color-white;
        cursor: pointer;
        font-size: 25px;
        border: 0;
    }
    &_content{
        text-align: center;
        margin: 30px 0;
        color: $color-hardcoal;
        font-weight: 500;
    }
    &_name{
        text-transform: capitalize;
        margin: 0;
        font-size: 32px;
        color: $color-black;
    }
    &_footer{
        display: flex;
        align-items: center;
        justify-content: space-between;
        padding-bottom: 30px;
        width: 60%;
        margin: 0 auto;

        &-cancel, &-confirm{
            cursor: pointer;
            font-weight: 700;
            width: 120px;
            height: 40px;
            background-color: $color-black;
            color: $color-white;
            border-radius: 30px;
            font-size: 22px;
        }
    }
}