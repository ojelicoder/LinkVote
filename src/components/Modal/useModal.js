import { useState } from 'react';

const useModal = () => {
    const [showModal, setShowModal] = useState(false),

    toggleModal = () => {
        setShowModal(!showModal);
    }

    return {toggleModal, showModal}
}

export default useModal;