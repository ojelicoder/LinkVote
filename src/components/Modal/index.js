import './modal.scss';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faTimes } from '@fortawesome/free-solid-svg-icons';

const Modal = (props) => {
    const {title, content, toggle, text, confirm} = props;

    return (
        <div className='modal_backdrop'>
            <div className='modal_container'>
                <header className='modal_header'>
                    <h3 className='modal_header-title'>{title}</h3>
                    <button className='modal_close' onClick={toggle}><FontAwesomeIcon icon={faTimes} /></button>
                </header>
                <main className='modal_content'>
                    {content}
                    <h2 className='modal_name'>{text}</h2>
                </main>
                <footer className='modal_footer'>
                    <button className='modal_footer-confirm' onClick={confirm}>OK</button>
                    <button className='modal_footer-cancel' onClick={toggle}>Cancel</button>
                </footer>
            </div>
        </div>
    )
}

export default Modal;