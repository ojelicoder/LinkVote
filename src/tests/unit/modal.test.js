import React from 'react';
import reactDom from 'react-dom';
import Modal from '../../components/Modal';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';

configure({ adapter: new Adapter() });

const modalProp = {
    title : 'Test Title',
    content: 'Test content text',
    text: 'Test',
    toggle: true,
    confirm: true
},

wrapper = shallow(<Modal props={modalProp} />);

describe('Modal', () => {

    it('renders without crashing', () => {
        const div = document.createElement('div');
        reactDom.render(<Modal props={modalProp} />, div);
    })

    it('renders DOM elements correctly', () => {
        const components = wrapper.find('div.modal_container');
        expect(components.length).toBe(1);
    })
    
    it('matches snapshot', () => {
        const tree = renderer.create(<Modal modalProp={{title: 'Matches title'}} />).toJSON();
        expect(tree).toMatchSnapshot();
    })

})