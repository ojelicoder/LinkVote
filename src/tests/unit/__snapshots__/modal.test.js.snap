// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal matches snapshot 1`] = `
<div
  className="modal_backdrop"
>
  <div
    className="modal_container"
  >
    <header
      className="modal_header"
    >
      <h3
        className="modal_header-title"
      />
      <button
        className="modal_close"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-times fa-w-11 "
          data-icon="times"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 352 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </button>
    </header>
    <main
      className="modal_content"
    >
      <h2
        className="modal_name"
      />
    </main>
    <footer
      className="modal_footer"
    >
      <button
        className="modal_footer-confirm"
      >
        OK
      </button>
      <button
        className="modal_footer-cancel"
      >
        Cancel
      </button>
    </footer>
  </div>
</div>
`;
